src/mlpack/core/util/version.hpp

```diff
@@ -18,7 +18,7 @@
 // with higher number than the most recent release.
 #define MLPACK_VERSION_MAJOR 4
 #define MLPACK_VERSION_MINOR 0
-#define MLPACK_VERSION_PATCH 1
+#define MLPACK_VERSION_PATCH 2
```

Comment:
@rcurtin I might be missing something here.  Since this PR is for releasing mlpack 4.0.1, is the patch level meant to be 1 or 2 ?

Reply:
Actually this is correct---what @mlpack-bot will do is tag HEAD~1 as the release (that's this commit: https://github.com/mlpack/mlpack/pull/3343/commits/a83346e6f939191c5560cc92f5c213c3980bbede).  So, `MLPACK_VERSION_PATCH` is `1` there.  It's only in the following commit that we bump it to `2`, so that `master` always has a version greater than the current stable version. :+1: